libs/auth/user.go

```diff
@@ -4,12 +4,21 @@ import (
 	"strings"
 
 	"github.com/databricks/cli/libs/textutil"
+	"github.com/databricks/databricks-sdk-go/service/iam"
 )
 
 // Get a short-form username, based on the user's primary email address.
 // We leave the full range of unicode letters in tact, but remove all "special" characters,
 // including dots, which are not supported in e.g. experiment names.
-func GetShortUserName(emailAddress string) string {
-	local, _, _ := strings.Cut(emailAddress, "@")
+func GetShortUserName(user *iam.User) string {
+	var name string
+	if IsServicePrincipal(user.UserName) {
+		// Try use the display name of the principal (if it has one)
+		name = user.DisplayName
+	}
+	if name == "" {
+		name = user.UserName
+	}
+	local, _, _ := strings.Cut(name, "@")
```

Comment:
I get what's happening but it reads backwards. Could you flip this around and assign the SP's `DisplayName` only if it is non-empty, instead? Then it captures the intent in 1 statement.